### ISSUE TITLE
Targeting net5 using net6 SDK causes pack failure

### ISSUE THREAD

**Issue by stefanloerwald** [dotnet/sdk#19665]
To reproduce, create the empty files `Component.razor` and `Component.razor.css` (no content required).

As the csproj file use

```xml
<Project Sdk="Microsoft.NET.Sdk.Razor">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <Nullable>enable</Nullable>
  </PropertyGroup>


  <ItemGroup>
    <SupportedPlatform Include="browser" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Components.Web" Version="5.0.9" />
  </ItemGroup>

</Project>
```

Then run `dotnet build` and afterwards `dotnet pack`. The latter will fail with the error message 

```log
C:\Program Files\dotnet\sdk\6.0.100-preview.7.21379.14\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.StaticWebAssets.5_0.targets(436,5):
error : Static web asset 'C:\project\obj\Debug\net5.0\scopedcss\projectbundle\RazorClassLibrary1.bundle.scp.css' has invalid source type ''. [C:\project\project.csproj]
```

This is using SDK 6.0.100-preview.7.21379.14.

**Comment by SonicGD** [comment]
We have the same issue. I tried daily build 6.0.100-rc.1.21411.30 but error still present.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
@javiercn can you please look into this? Thanks!

**Comment by javiercn** [comment]
This was fixed in RC1

**Comment by javiercn** [comment]
@stefanloerwald can you try a nightly build? This is a known limitation we introduced in preview7 as we were doing significant work in the area and we had to scope it for the milestone

**Comment by stefanloerwald** [comment]
Seems to work with rc1: https://github.com/stefanloerwald/dotnet-sdk-issue-19665/actions/runs/1189053349

**Comment by stefanloerwald** [comment]
It seems like the fix that was in RC1 previously was reverted? With current RC1, I can reproduce the issue again, see https://github.com/stefanloerwald/dotnet-sdk-issue-19665/runs/3618381368?check_suite_focus=true

When I fetch from the "daily" stream I get rc.1.21417.19, with which there is no issue. If I use 6.0.100-rc.1.21458.32, which to the best of my knowledge is the official RC1 version, I get the error.

**Comment by javiercn** [comment]
@stefanloerwald I think this was "partially" fixed on RC1 and is truly (hopefully) fixed for good in RC2.

Can you give it a try using RC2 nightly bits?

**Comment by stefanloerwald** [comment]
Hi @javiercn,

It is unclear to me how I could try RC2. When I last ran the workflow, the daily stream still gave me the RC1 version. The relevant workflow part is

```yml
 daily:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2.3.4
    - name: Setup .NET Core
      run: |
        wget https://dot.net/v1/dotnet-install.sh
        bash ./dotnet-install.sh --channel '6.0' --quality 'daily'
    - name: Install dependencies
      run: dotnet restore
    - name: Build
      run: dotnet build
    - name: Pack
      run: dotnet pack
``` 

If you could tell me how to adapt that for RC2, I'm happy to make the adjustments and let it run.

**Comment by javiercn** [comment]
@stefanloerwald something like this should do `.\dotnet-install.ps1 -Channel '6.0' -Version 6.0.100-rc.2.21465.13 -InstallDir .\dotnet-sdk-6.0.100-rc.2.21465.13` (note I'm using the powershell version)

**Comment by stefanloerwald** [comment]
Specifying the version seems to work in the workflow too. Seems like the issue is not in RC2: https://github.com/stefanloerwald/dotnet-sdk-issue-19665/runs/3621929948?check_suite_focus=true

Let's hope it stays like that...

**Comment by javiercn** [comment]
Yes, we don't plan to make changes in that area unless we really have to for the rest of 6.0. The initial partial fix was the result of a badly authored test that we improved in RC2 (along with a more robust fix), so hopefully things should be ok.

**Comment by javiercn** [comment]
@robertmclaws that build is from like a month ago, I'm not sure how many we retain. You can use the latest available RC2 version which is `6.0.100-rc.2.21505.1`
